### DISCUSSION TITLE
Controlling Yoetz decision making from inside Act systems

### DISCUSSION THREAD

**Discussion by idanarye** [idanarye/bevy-yoetz#9] [Ideas]
Yoetz makes a distinction between _Suggest_ systems - which are responsible for providing and scoring strategies - and _Act_ systems - which are responsible for enacting the strategy chosen by Yoetz.

In #5, @seivan argues that the active strategy should be given more control over the decision to prolong and/or replace it with a different strategy. With current (0.2) Yoetz, this can be achieved by having the Suggest system tweak the score - but that's a bit hacky, and also such things should be accessible from the Act systems which operate in the context of the action already running rather than the Suggest systems that typically look at the state of the world rather than the state of the AI.

**Comment by seivan** [comment]
I think there’s a misunderstanding. What I was talking about was lifecycles contained in a single place while still maintaining performance. I made a proof of concept I can show later.

**Reply by idanarye** [reply]
What do you mean by "single place"?
